### PR TITLE
*Added login functionality

### PR DESCRIPTION
**Pull request by maxonchil**
*Remaking field validation(disable button is some field is not valid)
*Start to working on separate servise for http handles

### CODE REVIEW

File: homework3/semi-uber/Server/api/routes/login.router.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+router.post("/", async (req, res) => {
+  try {
+    const { password, username } = req.body;
+    const user = await User.findOne({ username });
+    const token = await jwt.sign(user.id, secret);
+    const verify = await bcrypt.compare(password, user.password);
+
+    if (!verify) {
+      throw new Error();
```
Comment (MaxHouz) [inline homework3/semi-uber/Server/api/routes/login.router.js]: again, do not trow errors everywhere just send the user appropriate message and status code and that will be enough

File: homework3/semi-uber/Server/api/routes/login.router.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    logger.info(user);
+    res.status(200).json({ id: user.id, token: token });
+  } catch (err) {
+    res.status(400).json({ status: err.name });
+    logger.error(err.name);
+    throw err;
```
Comment (MaxHouz) [inline homework3/semi-uber/Server/api/routes/login.router.js]: you've got `catch` that also throws an error, please reconsider the error throwing in the whole application and if you have ome questions, contact me

File: homework3/semi-uber/UI/src/app/login-page/login-page.component.html [hunk 3]
```diff
@@ -1 +1,17 @@
...
+    type="text"
+    placeholder="username"
+  />
+  <input
+    formControlName="password"
+    class="login__password"
+    type="password"
+    placeholder="password"
+  />
+  <button [disabled]="!loginForm.valid" class="login__button button" (click)="login()">
```
Comment (MaxHouz) [inline homework3/semi-uber/UI/src/app/login-page/login-page.component.html]: you can use `[disabled]="loginForm.invalid"` without `no`t operator

File: homework3/semi-uber/Server/api/routes/login.router.js (hunk 1 above)
Comment (maxonchil) [reply homework3/semi-uber/Server/api/routes/login.router.js]: Make it to get it catch, if password is not valid and then send response with status

**Pull request merged**